api/src/data_migration/transformation/transform_util.py

```diff
@@ -183,6 +285,116 @@ def transform_opportunity_summary(
     return target_summary
 
 
+def convert_opportunity_summary_applicant_type(
+    source_applicant_type: SourceApplicantType,
+    existing_applicant_type: LinkOpportunitySummaryApplicantType | None,
+    opportunity_summary: OpportunitySummary,
+) -> LinkOpportunitySummaryApplicantType:
+    log_extra = get_log_extra_applicant_type(source_applicant_type)
+
+    # NOTE: The columns we're working with here are mostly the primary keys
+    #       While we do support updates, that's really only going to affect
+    #       the last update user + timestamps. From checking the prod data
+    #       there are basically zero updates to this data (~5 occurred 10+ years ago)
+    if existing_applicant_type is None:
+        logger.info("Creating new applicant type record", extra=log_extra)
+
+    applicant_type = transform_applicant_type(source_applicant_type.at_id)
+
+    # The legacy ID is named differently in the forecast/synopsis tables
+    if isinstance(source_applicant_type, (TapplicanttypesForecast, TapplicanttypesForecastHist)):
```

Comment:
It might be cleaner to add a method (maybe an `@property`) to each of the classes (`TapplicanttypesForecast`, `TapplicanttypesForecastHist`,  `TapplicanttypesSynopsis`, and `TapplicanttypesSynopsisHist`) that returns the appropriate column. Then we wouldn't have to use `isinstance` here and have knowledge of the internals of each class.